File: go-ut-bench/datasets/java/java_code_files_self_contained/boundary/boundary_001.java
class SafeMath {
    public static double safeDivide(double numerator, double denominator) {
        if (denominator == 0) {
            throw new ArithmeticException("Cannot divide by zero");
        }
        if (numerator == 0) {
            return 0.0;
        }
        double result = numerator / denominator;
        if (result > Double.MAX_VALUE) {
            return Double.POSITIVE_INFINITY;
        }
        if (result < -Double.MAX_VALUE) {
            return Double.NEGATIVE_INFINITY;
        }
        return result;
    }
    
    public static int safeSqrt(int value) {
        if (value < 0) {
            throw new IllegalArgumentException("Cannot compute square root of negative number");
        }
        if (value == 0) {
            return 0;
        }
        return (int) Math.sqrt(value);
    }
}

File: go-ut-bench/datasets/go/go_code_files_self_contained/boundary/boundary_001.go
package main

func SafeDivide(numerator, denominator float64) float64 {
	if denominator == 0 {
		panic("cannot divide by zero")
	}
	if numerator == 0 {
		return 0.0
	}
	result := numerator / denominator
	return result
}

func SafeSqrt(value int) int {
	if value < 0 {
		panic("cannot compute square root of negative number")
	}
	if value == 0 {
		return 0
	}
	i := 0
	for i * i <= value {
		i++
	}
	return i - 1
}